cardinal/events/events_test.go

```diff
@@ -52,10 +56,92 @@ func TestEvents(t *testing.T) {
 				assert.Equal(t, mode, websocket.TextMessage)
 				assert.Equal(t, string(message)[:4], "test")
 				//fmt.Println(string(message))
+				count.Add(1)
 			}
 		}()
 	}
 	wg.Wait()
-	err := txh.Shutdown()
-	assert.NilError(t, err)
+	txh.EventHub.ShutdownEventHub()
+	assert.Equal(t, count.Load(), int32(numberToTest*numberToTest))
+}
+
+type garbageStructAlpha struct {
+	Something int `json:"something"`
+}
+
+func (garbageStructAlpha) Name() string { return "alpha" }
+
+type garbageStructBeta struct {
+	Something int `json:"something"`
+}
+
+func (garbageStructBeta) Name() string { return "beta" }
+
+type SendEnergyTx struct {
+	From, To string
+	Amount   uint64
+}
+
+type SendEnergyTxResult struct{}
+
+func TestEventsThroughSystems(t *testing.T) {
+	numberToTest := 5
+	w := ecs.NewTestWorld(t)
+	sendTx := ecs.NewTransactionType[SendEnergyTx, SendEnergyTxResult]("send-energy")
+	assert.NilError(t, w.RegisterTransactions(sendTx))
+	counter1 := atomic.Int32{}
+	counter1.Store(0)
+	for i := 0; i < numberToTest; i++ {
+		w.AddSystem(func(wCtx ecs.WorldContext) error {
+			wCtx.GetWorld().EmitEvent(&events.Event{Message: "test"})
+			counter1.Add(1)
+			return nil
+		})
+	}
+	assert.NilError(t, ecs.RegisterComponent[garbageStructAlpha](w))
+	assert.NilError(t, ecs.RegisterComponent[garbageStructBeta](w))
+	assert.NilError(t, w.LoadGameState())
+	txh := test_utils.MakeTestTransactionHandler(t, w, server.DisableSignatureVerification())
+	url := txh.MakeWebSocketURL("events")
+	dialers := make([]*websocket.Conn, numberToTest, numberToTest)
```

Comment:
```suggestion
	dialers := make([]*websocket.Conn, numberToTest)
```
you can drop the second arg here, first arg to make is length, second is capacity